public/app/features/alerting/unified/rule-list/hooks/useFilteredRulesIterator.ts

```diff
@@ -68,11 +68,16 @@ export function useFilteredRulesIteratorProvider() {
       map(([group, rule]) => mapGrafanaRuleToRuleWithOrigin(group, rule))
     );
 
-    const [source, ...iterables] = ruleSourcesToFetchFrom.map((ds) => {
-      return from(prometheusGroupsGenerator(ds, groupLimit)).pipe(map((group) => [ds, group] as const));
+    const sourceIterables = ruleSourcesToFetchFrom.map((ds) => {
+      const generator = prometheusGroupsGenerator(ds, groupLimit);
+      return from(generator).pipe(map((group) => [ds, group] as const));
     });
 
-    const dataSourcesIterator = merge(source, ...iterables).pipe(
+    // if we have no prometheus data sources, use an empty async iterable
+    const source = sourceIterables.at(0) ?? empty();
```

Comment:
root cause; de-structuring from the array means `source` can be `undefined` (TypeScript doesn't check this because reasons).

The `merge()` doesn't function when `undefined` is passed as the source so we use `empty()` instead to allow consumer to pull and not block anything / cause errors.